test/algorithms/test_bernsteinvazirani.cpp

```diff
@@ -143,3 +136,19 @@ TEST_P(BernsteinVazirani, DynamicEquivalenceSimulation) {
 
   EXPECT_NEAR(fidelity, 1.0, 1e-4);
 }
+
+TEST_F(BernsteinVazirani, BVBenchmarkSimulate) {
+  auto nqubits = 20;
+  auto qc = std::make_unique<qc::BernsteinVazirani>(nqubits);
+  qc::CircuitOptimizer::removeFinalMeasurements(*qc);
+  const auto out = dd::benchmarkSimulate(*qc);
+  EXPECT_NE(out->sim.p, nullptr);
+}
+
+TEST_F(BernsteinVazirani, BVBenchmarkFunctionality) {
+  auto nqubits = 20;
+  auto qc = std::make_unique<qc::BernsteinVazirani>(nqubits);
+  qc::CircuitOptimizer::removeFinalMeasurements(*qc);
+  const auto out = dd::benchmarkFunctionalityConstruction(*qc);
+  EXPECT_NE(out->func.p, nullptr);
+}
```

Comment:
These tests are redundant, aren't they? the behaviour is inherently already tested by the other methods.
Please also adjust for the other files. Anyhow: these tests do not test anything besides the fact that the simulation runs and produces _something_. This might give a false sense of security from the resulting good coverage.

Reply:
Furthermore, these tests are not parametrised for some reason.